lib/notification_banner.rb

```diff
@@ -0,0 +1,22 @@
+class NotificationBanner
+  include Singleton
+  attr_reader :message
+
+  def load_config
+    config = YAML.load(
```

Comment:
I know this code isn't ready for review yet, but just to flag up that this is a lot of custom code to do things in a custom way, when you could save a lot of lines of code and keep things more consistent if you follow the pattern for, e.g. inactive offices. So you just need one line in application.rb that says:

```
config.x.banners= config_for(:banners, env: ENV.fetch('ENV', 'localhost'))
```

Then in here you just need:

```
def active_banner
  Rails.configuration.x.banners.find do |banner|
    ( DateTime.parse(banner['date_from']..DateTime.parse(banner['date_to']).include?(DateTime.now)
  end
```

And then you're done. You don't need HostEnv, or YAML.load, or Singleton patterns or anything, and your banners.yml file can have a standardised format to handle different environments.

Reply:
Gotta love github and its fake single-page app liveness...

But yeah I agree, there's room to simplify here I think

Reply:
indeed this does simplify things a great deal - got stuck in a rabbit hole trying to understand what the best pattern was for Singleton classes and lost sight of other ways to do it